include/envoy/http/header_map.h

```diff
@@ -127,7 +127,7 @@ class HeaderString {
   /**
    * @return the modifiable backing buffer (either inline or heap allocated).
    */
-  char* buffer() { return buffer_.dynamic_; }
+  char* buffer() const { return buffer_.dynamic_; }
```

Comment:
erm, this is returning modifiable buffer so isn't really `const`?

Reply:
It's pretty standard to have these two:

```
char* buffer() { return buffer_.dynamic_; }
const char* buffer() const { return buffer_.dynamic_; }
```

Though I'm curious why we would need to return a non-const `char*` in this case.

Reply:
Yep found it. It was in a test that was calling buffer on a const `HeaderString`. It shouldn't be used in that scenario since yah it's a modifiable buffer.

We do still need the non-const buffer though, since in the codecs we construct a `HeaderString`, modify it, and then move it in to the HeaderMap.